signals/generators/ios/core_data.py

```diff
@@ -199,12 +200,33 @@ def get_model(xml):
     return new_model
 
 
-def write_xml_to_file(xml, objects):
-    model = get_model(xml)
+# parses the hidden .xccurrentversion for the name of the current version
+# this information is stored inside its first and only <string> xml tag
+def get_current_version(xcdatamodeld_path):
+    xccurrentversion_path = xcdatamodeld_path + '/.xccurrentversion'
+
+    # .xccurrentversion might not exist if there is only one version
+    if os.path.exists(xccurrentversion_path):
+        xml_dom = minidom.parse(xccurrentversion_path)
+        return xml_dom.getElementsByTagName('string')[0].childNodes[0].data
+    else:
+        return [f for f in os.listdir(xcdatamodeld_path) if f.endswith('.xcdatamodel')][0]
+
+
+def get_core_data_from_folder(xcdatamodeld_path):
+    current_version_name = get_current_version(xcdatamodeld_path)
+    core_data_path = xcdatamodeld_path + '/' + current_version_name + '/contents'
```

Comment:
might be better to use the python string formatting instead of appending together with `+`